lib/multicell_location/services/nrf_cloud_integration.c

```diff
@@ -143,13 +143,14 @@ int location_service_get_cell_location_nrf_cloud(
 	struct nrf_cloud_rest_context rest_ctx = {
 		.connect_socket = -1,
 		.keep_alive = false,
-		.timeout_ms = NRF_CLOUD_REST_TIMEOUT_NONE,
+		.timeout_ms = (params->timeout > 0) ? params->timeout * 1000 :
+						      NRF_CLOUD_REST_TIMEOUT_NONE,
```

Comment:
the value of `NRF_CLOUD_REST_TIMEOUT_NONE` should be changed to `0` correct?

Reply:
That probably makes sense. I'm doing the change and let's see if we'll find a problem with it.

Reply:
I will still need to investigate what zero does within http_client library in Zephyr

Reply:
`http_client` doesn't initialize its timeout handler when timeout is `K_NO_WAIT` (which is 0). However, API description says zero shouldn't be used:
`The timeout value cannot be 0 as there would be no time to receive the data.`

The API doesn't say what the value should be that it wouldn't start the timeout but `SYS_FOREVER_MS` and `0` will not start the timeout handler.
Right now I'm kind of leaning towards keeping the implementation as it is now. The other option would be in my opinion that we would use `SYS_FOREVER_MS` in several components: rest_client, nrf_cloud_rest, multicell_location and location library. It would be possible also to just change rest_client to pass SYS_FOREVER_MS to http_client when rest_client gets zero timeout but I don't know if it's any better than zero given it's not mentioned in the API.
If timeout is not passed to http_client, behavior is then server dependent on when it resets or closes the TCP connection.

What do you think @jhirsi, @tokangas, @jayteemo?

Reply:
ah, i missed that you are still passing `timeout_ms` to the http_client.
on first glance, i thought this PR was consolidating the timeouts to only use 'timeout_ms` for the socket and to not use the http_client timeout mechanism.

changing `NRF_CLOUD_REST_TIMEOUT_NONE` to `0` is not necessary if the http_client timeout is still being used.

does the http_client timeout mechanism work correctly?

Reply:
> does the http_client timeout mechanism work correctly?

Yes, it works fine although the timeout in http_client is only for data receiving (not for sending). The timeout is not set if given timeout is `SYS_FOREVER_MS` or `0`.

Reply:
> changing `NRF_CLOUD_REST_TIMEOUT_NONE` to `0` is not necessary if the http_client timeout is still being used.

I think I'd like to change `NRF_CLOUD_REST_TIMEOUT_NONE` to `0` just to align with rest_client, multicell_location library and location library.

Reply:
I will close this comment as we have changed this back to SYS_FOREVER_MS.